### ISSUE TITLE
[TODO]: Better documentation for common Q&As and also for developers

### ISSUE THREAD

**Issue by oscarwumit** [chemprop/chemprop#770]
**Notes**
From time to time we get similar issues from different users, so it seems to be helpful to create a Q&A section in the documentation page or somewhere else. Examples of commonly asked questions are (not comprehensive and we should keep a record):
- Installation chemprop properly with GPU (sometimes users installed the CPU version without realizing it, and subsequently complain about the training time to be really long)
- Several users have asked about general best practices and guidance to pick hyper-parameters and settings for different tasks (regression, classification) and on how to best plan out hyper-parameter optimization in general.
- How to control the randomness of chemprop, e.g., difference between seed, pytorch_seed, and other randomness in the program.

Also, sometimes we implement certain features meant for developer use only, and we should document the details. For example: we might want to recommend the developer to do developmental install (pip install -e option) and install pytest when setting up chemprop environment. Also, certain test might only work on CPU and might fail in GPU environment, and we should note that for the developers.

**Comment by kevingreenman** [comment]
Maybe we should keep developer docs in our existing [private repo](https://github.com/chemprop/chemprop-developers/tree/main) unless we're thinking of having many people from outside our group becoming active developers? Or we could also try to separate things that anyone making a PR might need to know from things that only internal people should know.

**Comment by oscarwumit** [comment]
That could work for the developer docs.
